client/blocks/product-selector/index.jsx

```diff
@@ -114,15 +136,19 @@ export class ProductSelector extends Component {
 		const { title, optionDisplayNames } = product;
 		const purchase = this.getPurchaseByProduct( product );
 
-		if ( ! purchase ) {
-			return title;
+		// Product display name, obtained from a purchased product
+		if ( purchase && optionDisplayNames && optionDisplayNames[ purchase.productSlug ] ) {
+			return optionDisplayNames[ purchase.productSlug ];
 		}
 
-		if ( ! optionDisplayNames || ! optionDisplayNames[ purchase.productSlug ] ) {
-			return title;
+		// Product display name, obtained from a product that's included in a purchased plan
+		const planProductSlug = this.getProductSlugByCurrentPlan();
+		if ( planProductSlug && optionDisplayNames[ planProductSlug ] ) {
```

Comment:
Should we also check if `optionDisplayNames` array exists before trying to access `optionDisplayNames[ planProductSlug ]`?

I'm asking since we're checking this a [few lines above](https://github.com/Automattic/wp-calypso/pull/37396/files#diff-dfff688a37f93f6dc6effc4f8b83730dR140) and not here.

Reply:
👍 Fixed in https://github.com/Automattic/wp-calypso/pull/37396/commits/a123b5f15c4a6fd0b5321dd995f3328b388678a8.